### PR TITLE
documentation: agent: add flows for 4.7.6 test

### PR DESCRIPTION
**Pull request by mariomaz**
Create a UML diagram to reflect the design of the Multi-AP Policy
Config Request message flow implementation.

Create a UML diagram of the 4.7.6 flow "Per-AP Metrics Response
Controlled by AP Metrics Channel Utilization Reporting Threshold test"

Update ap_metrics.md to describe and refer to those UML diagrams.

### CODE REVIEW

File: documentation/images/plantuml/agent/flows/ap_metrics_cert_4.7.6.puml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@startuml
+participant controller
```
Comment (arnout) [inline documentation/images/plantuml/agent/flows/ap_metrics_cert_4.7.6.puml:2]: I didn't know about these declarations! #til

File: documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+@startuml
+controller->backhaul_manager: send_cmdu\n(MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE)
+
+backhaul_manager->backhaul_manager: handle_multi_ap_policy_config_request()\nStore AP Metrics Reporting Interval\nvalue for periodic AP metrics reporting.\nForward CMDU to all slaves whose RUID is\nspecified in Metric Reporting Policy TLV\n
+backhaul_manager->son_slave_wlan0: forward_cmdu\n(MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE)
+backhaul_manager->son_slave_wlan2: forward_cmdu\n(MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE)
```
Comment (arnout) [inline documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml:6]: (nitpick) Although technically inaccurate, for the AP Metrics Query flow we have written it down as if it were synchronous, i.e. first the ACK, then the full flow for wlan0, then the full flow for wlan2. We felt that this was visually more easy to understand.

File: documentation/flows/agent/ap_metrics.md [hunk 3]
```diff
@@ -5,22 +5,23 @@ The AP metrics can optionally include station metrics and statistics, as configu
...
+Access Point Metrics should be represented as 3 messages:
 
-* AP Metrics query
-* AP Metrics response
+* AP Metrics Query
+* AP Metrics Response
+* Multi-AP Policy Config Request
 
-## AP Metrics query
+## AP Metrics Query
```
Comment (arnout) [inline documentation/flows/agent/ap_metrics.md:14]: Good catch!

File: documentation/flows/agent/ap_metrics.md [hunk 4]
```diff
@@ -5,22 +5,23 @@ The AP metrics can optionally include station metrics and statistics, as configu
...
 TLV payload are:
 
 * List of BSSIDs for which the metrics are to be reported.
 
-## AP Metrics response
+## AP Metrics Response
 
-`AP Metrics response` message has type `0x800c`.
+`AP Metrics Response` message has type `0x800c`.
 It has one or more AP Metric TLVs with type `0x94` (one per BSSID reported).
```
Comment (arnout) [inline documentation/flows/agent/ap_metrics.md]: While you're correcting things: it's actually called AP Metric***s*** TLV. I made that mistake :-)

File: documentation/flows/agent/ap_metrics.md [hunk 5]
```diff
@@ -46,7 +66,20 @@ Then it forwards the request to these slave(s).
...
+
+Multi-AP Policy Config Request is originated by the controller.
+Agent processes the request and responds with an ACK Message.
+
+![UML](../../images/plantuml/agent/flows/multi_ap_policy_config_request.png)
+
+The Multi-AP Policy Config Request message is received by the backhaul manager.
+The backhaul manager stores AP Metrics Reporting Interval value for periodic AP metrics reporting and forwards received message "as is" to all slaves whose RUID is specified in Metric Reporting Policy TLV
+The son_slave simply forwards to the monitor.
+The monitor stores configuration provided in Metric Reporting Policy TLV into monitor database.
```
Comment (arnout) [inline documentation/flows/agent/ap_metrics.md]:
```suggestion
The monitor stores the radio-specific configuration provided in the Metric Reporting Policy TLV into monitor database if the radio UID matches.
```

File: documentation/flows/agent/ap_metrics.md [hunk 6]
```diff
@@ -62,16 +95,39 @@ AP Metrics Query should pass the following verifications:
...
+* Verify that the agent includes one Associated STA Traffic Stats TLV for the associated STA.
+* Verify that the Associated STA Traffic Stats TLV is decodable and includes all fields per TLV definition in [1]
+* Verify that the agent includes one Associated STA Link Metrics TLV for the associated STA.
+* Verify that the Associated STA Link Metrics TLV is decodable and includes all fields per TLV definition in [1]
+
+
+## 4.7.6 MAUT Per-AP Metrics Response Controlled by AP Metrics Channel Utilization Reporting Threshold test
+
+This test verifies that a MAUT correctly sends AP and STA related metrics in the AP Metrics Response message based on the AP Metrics Channel
+Utilization Reporting Threshold setting specified by the Controller in the Multi-AP Policy Config Request message.
```
Comment (arnout) [inline documentation/flows/agent/ap_metrics.md]: (nitpick) CODING_STYLE.md specifies one line per sentence.

```suggestion
This test verifies that a MAUT correctly sends AP and STA related metrics in the AP Metrics Response message based on the AP Metrics Channel Utilization Reporting Threshold setting specified by the Controller in the Multi-AP Policy Config Request message.
```

More importantly, though, I think the following additional sentences could help a lot to clarify what the test really does.

```
After setting the policy, a STA is connected to the agent and starts sending traffic.
This causes the channel utilisation to increase and exceed the configured threshold.
At that point, the agent must send an AP Metrics Response message to the controller.
```

**Review by arnout — APPROVED** [review]
Really excellent. I believe this documentation will really pay off. If we had this for all the existing flows, it would be a *lot* easier to dive into the code. And even for people who are already familiar it helps a lot to read this rather than trying to follow all the steps in the code.

File: documentation/flows/agent/ap_metrics.md (hunk 4 above)
Comment (mariomaz) [reply documentation/flows/agent/ap_metrics.md]: Fixed

File: documentation/flows/agent/ap_metrics.md (hunk 5 above)
Comment (mariomaz) [reply documentation/flows/agent/ap_metrics.md]: Done

File: documentation/flows/agent/ap_metrics.md (hunk 6 above)
Comment (mariomaz) [reply documentation/flows/agent/ap_metrics.md]: Done

File: documentation/images/plantuml/agent/flows/ap_metrics_cert_4.7.6.puml (hunk 1 above)
Comment (mariomaz) [reply documentation/images/plantuml/agent/flows/ap_metrics_cert_4.7.6.puml:2]: I fell in love with PlantUML the first time I knew about. However, I decided not to memorize its syntax at all. When I have to create a diagram, I always open https://plantuml.com/ side by side and use the examples as a guide.

File: documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml (hunk 2 above)
Comment (mariomaz) [reply documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml:6]: I was going to propose you the modification of AP Metrics Query flow to make it more accurate!

**Review by rmelotte — APPROVED** [review]
The commit headline of the fixup commit looks wrong.
Other than that, LGTM!

File: documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml (hunk 2 above)
Comment (arnout) [reply documentation/images/plantuml/agent/flows/multi_ap_policy_config_request.puml:6]: Either way works for me. It's a matter of taste I guess. Consistency would probably be good. But unified agent will anyway need to change this again, so it's not very important I supposed. Maybe having two different examples is actually good - when someone new reads it, they can decide which option they like most :-)

**Pull request merged**